test/loco/constraints/test_utils.clj

```diff
@@ -46,33 +46,29 @@
        ))
      ~msg)))
 
-(defmacro test-loco [input expected]
-  (let [{:keys [identity model compiled solutions]} expected]
-    `(let [
-           input#     ~input
-           expected#  ~expected
-           identity#  ~identity
-           model#     ~model
-           compiled#  ~compiled
-           solutions# ~solutions
-           model-fn#     (memoize ~model/compile)
-           compile-fn#   (memoize ~compiler/compile)
-           solutions-fn# ~solver/solutions
-           compiled-strings# (juxt
-                              (comp (partial mapv str) :vars)
-                              (comp (partial mapv str) (memfn ~'getCstrs) :model))
-           ]
-       (assert (every? #{:identity :model :compiled :solutions} (keys expected#)))
-       (when identity# (is (= identity#
-                              input#)
-                           "identity"))
-       (when model# (is (= model#
-                           (model-fn# input#))
-                        "model"))
-       (when compiled# (is (= compiled#
-                              (->> input# model-fn# compile-fn# compiled-strings#))
-                           "compiled"))
-       (when solutions# (is (= solutions#
-                               (->> input# model-fn# compile-fn# solutions-fn# set))
-                            "solutions"))       
-       )))
+(defmacro test-loco
+  {:style/indent :defn}
+  ([input expected] `(test-loco nil ~input ~expected))
+  ([name input expected]
+   (let [{:keys [identity model compiled solutions]} expected
+         model-fn     (gensym "model-fn")
+         compile-fn   (gensym "compile-fn")]
+     (assert (every? #{:identity :model :compiled :solutions} (keys expected)))
+     `(let [~model-fn     model/compile
```

Comment:
the `model-fn#` syntax has the same effect as the gensym usage. possibly this is a clojure idiom.

Reply:
`model-fn#` is a different gensym in each backquoted form, so I had to switch to gensym when I raised the `when` statements to run at macro time. Generally I wanted errors to contain the actual info rather than just macro-internal gensym names, which is why I did this refactor.